artifacts/CMakeLists.txt

```diff
@@ -0,0 +1,115 @@
+#     "base").
+#   - {dist_bundle_name}: For slices that are AMDGPU target specific, this is
+#     the value of -DTHEROCK_AMDGPU_DIST_BUNDLE_NAME, which by default is
+#     derived from the gfx name (e.g. "gfx1100") for single target slices or
+#     the family name for multi-target slices (e.g. "gfx103X-igpu"). See
+#     therock_amdgpu_targets.cmake for derivations. The specific set of built
+#     target and family slices is defined by the relevant CI setup (e.g. the
+#     presubmit/smoketest pipeline may build a small multi-arch family for
+#     basic validation, whereas the full test/postsubmit pipeline may build
+#     families in parallel that cover all supported targets for nightly
+#     channels, etc, and the weekly full channel may build bundles of all
+#     supported architectures in one). We generally partition this in order to
+#     match CI and testing fan-out parallelism needed in order to meet CI
+#     latency SLAs for different channels.
+#   - {variant}: The build variant. For normal release builds this will be
+#     "release". Various sanitizer and debug variants are also defined. Not
+#     all combinations are distinct (i.e. the "doc" component may be the same
+#     for all santizer variants), but we assemble and publish them regardless.
+#   - {component}: For distributed C++ artifacts, there will typically be a
+#     standard set of components (e.g. "runfiles", "dev", "doc", "dbg", etc).
+#     Different pipelines can include just the slices they need, reducing
+#     CI churn and overhead. See `Components` below for more definition.
+#
+# Slices
+# ------
+# The standard ROCm build consists of the following slices:
+#   - "third-party": stage directories of all needed third-party libraries.
+#     All slices can be considered to conditionally depend on third-party.
+#     Whether they do actually or not is build configuration dependent (i.e.
+#     some builds may be configured to use system vs bundled deps, etc).
+#     Dependencies: None
+#     Target specifity: Target neutral
+#
+#   - "base": stage directories for sub-projects in the base/ directory.
+#     Dependencies: None
+#     Target specifity: Target neutral
+#
+#   - "compiler-amd-llvm": The compiler/amd-llvm sub-project.
+#     Dependencies: base
+#     Target specificity: Target neutral
+#
+#   - "compiler-hipify": The compiler/hipify sub-project.
+#     Dependencies: base, compiler-amd-llvm
+#     Target specificity: Target neutral
+#
+#   - "core-runtime": Core sub-projects that do not have a runtime dependency
+#     on the compiler (e.g. ROCR-Runtime, rocminfo, etc).
+#     Dependencies: base
+#     Target specificity: Yes
+#
+#   - "core-hip": HIP runtime components, including compiler interfaces.
+#     Dependencies: base, core-runtime, compiler-amd-llvm
+#     Target specificity: Yes
+#
+#   - "comm-libs-rccl": The RCCL sub-project.
+#     Dependencies: base, core-hip, compiler-hipify
+#     Target specificity: Yes
```

Comment:
Is there a (public) chart showing this dependency graph? Should TheRock include one? Are there any dependency links that we want to rework as part of this (e.g. decoupling components)?

Reply:
Stella doesn't draw :)

But there probably should be.